### ISSUE TITLE
Oncoplot customization

### ISSUE THREAD

**Issue by Sebbyyy** [PoisonAlien/maftools#490]
I was just wondering as I don't know if I am capable of customizing the oncoplot function myself and others may find this useful if you could include a couple of customization options?

- Could you allow the options to include other types of CNVs? I am trying to include Shallow as an alteration but the function does not recognize it. May be useful to be able to customize the CNV names?

- Could you allow an option to allow actual mutations as the little boxes and the CNVs as the filled columns to mimic plots made by the TCGA?

**Comment by PoisonAlien** [comment]
Hello,
Thanks for the requests. I will try my best but second point is unusally tricky to achieve using base R graphics. I will see what I can do.

**Comment by Sebbyyy** [comment]
I may be very wrong here and i apologize but in my head, couldn't you just assign the mutations to the same properties that the CNV has now and visa versa for the second point? As its just reversing the representations for the CNVs and mtuations.

**Comment by PoisonAlien** [comment]
If I understand correctly, its just adding borders to mutations and CNVs? Can you post an example or a screencap of TCGA?

**Comment by Sebbyyy** [comment]
![image](https://user-images.githubusercontent.com/40489027/78340025-c2cfb780-75d8-11ea-870e-1837939bf979.png)

Like that? So it mimics the output you can get from TCGA

**Comment by tashbroit111** [comment]
Hi,
I'm working on this same project as Sebbyy and have just discovered that it accepts "Loss" instead of "Shallow" as an input, which is perfectly fine. However, it doesn't treat the call in the same way, i.e. the way it fills the column bar is as it would a mutation and not a copy number change. If you're able to adjust the way the columns are colour-filled, can loss be made to be the same as other copy number alterations? Example of what it looks like currently in the image:

![Capture](https://user-images.githubusercontent.com/64013001/79746595-955f6980-834d-11ea-89fb-1688e7bd9177.JPG)

**Comment by PoisonAlien** [comment]
Could you reinstall it from GitHub and try again? I have recently fixed this grid width issue with CNVs.

**Comment by tashbroit111** [comment]
That's excellent news - thank you very much! I'm new to R and am using R Studio. To update the package, can I just update using BiocManager::install("maftools")? Or is what you're suggesting a different process?

**Comment by PoisonAlien** [comment]
You need to install from GitHub because these changes are not available on Bioconductor yet.
```r
BiocManager::install("PoisonAlien/maftools")
```

**Comment by tashbroit111** [comment]
I got it installed and the new modifications to CNV look great. However it seems to only be applying the new grid width to "Amp" and "Del" copy number calls. Are you able to extend this new width formatting to "Loss" calls? The image shows del in red and loss in orange as it currently stands:

![Capture](https://user-images.githubusercontent.com/64013001/79753588-88e10e00-8359-11ea-849b-c6414e687fcb.JPG)

**Comment by PoisonAlien** [comment]
Hi,
I am sorry for the delay. I have pushed some changes which should take care of any types of CN values. Could you please check it and let me know if it works for you?

**Comment by Sebbyyy** [comment]
@PoisonAlien Thanks mate.

I will let @tashbroit111 have a look first and let you know.

**Comment by tashbroit111** [comment]
Hi @PoisonAlien - I just ran my script again and it looks like it's applying the colour fill equally among to CN calls. I've popped in an example below. It looks great and I really appreciate this adjustment to the code. Thank you!

Image: orange = copy loss, red = hom del and blue = missense.

![image](https://user-images.githubusercontent.com/64013001/81038914-e6916080-8eea-11ea-999b-92b7d4552cd2.png)
